src/main/java/com/team701/buddymatcher/services/pairing/impl/PairingServiceImpl.java

```diff
@@ -1,19 +1,57 @@
 package com.team701.buddymatcher.services.pairing.impl;
 
+import com.team701.buddymatcher.domain.users.Buddies;
+import com.team701.buddymatcher.domain.users.User;
+import com.team701.buddymatcher.repositories.users.BuddiesRepository;
+import com.team701.buddymatcher.repositories.users.UserRepository;
 import com.team701.buddymatcher.services.pairing.PairingService;
+import com.team701.buddymatcher.services.users.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import javax.transaction.Transactional;
+import java.util.List;
+
 /**
  * Implementation for the PairingService which is a service providing the implementations of the methods
  * for the pairing endpoints
  */
 @Service
 public class PairingServiceImpl implements PairingService {
 
+    private final UserRepository userRepository;
+    private final BuddiesRepository buddiesRepository;
+
+    private final UserService userService;
+
+    @Autowired
+    public PairingServiceImpl(UserRepository userRepository, BuddiesRepository buddiesRepository, UserService userService) {
+        this.userRepository = userRepository;
+        this.buddiesRepository = buddiesRepository;
+        this.userService = userService;
+    }
     @Override
+    @Transactional
     public void addBuddy(Long userId, Long buddyId) {
-        //Currently just a blank implementation for testing endpoint call
-        System.out.println(String.format("Buddy add request: %s, %s",userId,buddyId));
+        User user = userService.retrieve(userId);
```

Comment:
userService.retrieve(userId), is the old method name, a new change just went out and you need to use `userService.retrieveById(userId)` to get a user object from the user repo. Try rebase your branch on main and it should come up

Reply:
Yep thanks for noticing that, I think the problem has been solved now.